### PR TITLE
2.x: Fix Generic issue in Single

### PR DESCRIPTION
**Pull request by vanniktech**

### CODE REVIEW

File: src/main/java/io/reactivex/Single.java [hunk 1]
```diff
@@ -874,7 +874,7 @@
      * @return the new Single instance
      * @since 2.0
      */
-    public static <T, R> Single<R> zip(final Iterable<? extends SingleSource<? extends T>> sources, Function<? super Object[], ? extends R> zipper) {
+    public static <T, R> Single<? extends R> zip(final Iterable<? extends SingleSource<? extends T>> sources, Function<? super Object[], ? extends R> zipper) {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Single.java:877]: We never return `? extends X` as it adds burden to the receiver. Why did you do this?

Comment (vanniktech) [reply src/main/java/io/reactivex/Single.java:877]: Otherwise I get a compile error in IntelliJ

Comment (akarnokd) [reply src/main/java/io/reactivex/Single.java:877]: Could be a bug with IntelliJ as javac and Eclipse was fine with the original.

Comment (vanniktech) [reply src/main/java/io/reactivex/Single.java:877]: Yeah I also have no problems when running `./gradlew test`

Comment (akarnokd) [reply src/main/java/io/reactivex/Single.java:877]: Are you using the latest IntelliJ? A possible workaround could be that you put in `return (Single<R>)...` and then suppress the invalid cast.

Comment (vanniktech) [reply src/main/java/io/reactivex/Single.java:877]: Not quite the latest. I'm running Android Studio beta2. So should I change this PR to cast the value and suppress the warning?

Comment (akarnokd) [reply src/main/java/io/reactivex/Single.java:877]: If this error prevents you from working further on RxJava then do the changes.

Comment (vanniktech) [reply src/main/java/io/reactivex/Single.java:877]: Nah I can live with terminal for executing tests

**Pull request closed without merging**